src/components/dialog/dialog.ts

```diff
@@ -402,8 +387,7 @@ export class SbbDialogElement extends LitElement {
   }
 
   protected override render(): TemplateResult {
-    const hasTitle = !!this.titleContent || this._namedSlots['title'];
-    const hasActionGroup = this._namedSlots['action-group'] && hasTitle;
+    const hasTitle = !!this.titleContent || this._namedSlots.slots.has('title');
```

Comment:
Again, nice catch!
SlotChange does not trigger a new rendering.
In case a slotted title would be added/removed while the dialog is open, we would be in a faulty state.